metrics/cmd/heapster-apiserver/app/heapstermetrics.go

```diff
@@ -21,19 +21,45 @@ import (
 	metricsink "github.com/kubernetes-incubator/metrics-server/metrics/sinks/metric"
 	nodemetricsstorage "github.com/kubernetes-incubator/metrics-server/metrics/storage/nodemetrics"
 	podmetricsstorage "github.com/kubernetes-incubator/metrics-server/metrics/storage/podmetrics"
+	"k8s.io/apimachinery/pkg/apimachinery/announced"
+	"k8s.io/apimachinery/pkg/apimachinery/registered"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/apiserver/pkg/registry/rest"
 	genericapiserver "k8s.io/apiserver/pkg/server"
 	v1listers "k8s.io/client-go/listers/core/v1"
-	"k8s.io/client-go/pkg/api"
 	"k8s.io/metrics/pkg/apis/metrics"
-	_ "k8s.io/metrics/pkg/apis/metrics/install"
+	"k8s.io/metrics/pkg/apis/metrics/install"
 	"k8s.io/metrics/pkg/apis/metrics/v1alpha1"
 )
 
+var (
+	groupFactoryRegistry = make(announced.APIGroupFactoryRegistry)
+	registry             = registered.NewOrDie("")
+	Scheme               = runtime.NewScheme()
+	Codecs               = serializer.NewCodecFactory(Scheme)
+)
+
 func installMetricsAPIs(s *options.HeapsterRunOptions, g *genericapiserver.GenericAPIServer,
 	metricSink *metricsink.MetricSink, nodeLister v1listers.NodeLister, podLister v1listers.PodLister) {
+	install.Install(groupFactoryRegistry, registry, Scheme)
```

Comment:
@sttts @deads2k I know in another PR, there was concern about having a function mutate a package-level variable such as `Scheme`. Could you recommend some best practices here?

- in which file should `Scheme` and `Codecs` live?
- is something like this ok, or do we need to put this in an `init()` function? (note, I dislike `init()` magic, and would prefer to avoid it if possible.)